pkg/clustermanager/cluster_manager_test.go

```diff
@@ -615,11 +615,9 @@ func TestClusterManagerUpgradeSelfManagedClusterSuccess(t *testing.T) {
 	kcp, mds := getKcpAndMdsForNodeCount(0)
 	tt := newSpecChangedTest(t)
 	tt.mocks.client.EXPECT().GetEksaCluster(tt.ctx, tt.cluster, tt.clusterSpec.Cluster.Name).Return(tt.oldClusterConfig, nil)
-	tt.mocks.client.EXPECT().GetBundles(tt.ctx, tt.cluster.KubeconfigFile, tt.cluster.Name, "").Return(test.Bundles(t), nil)
-	tt.mocks.client.EXPECT().GetEksdRelease(tt.ctx, gomock.Any(), constants.EksaSystemNamespace, gomock.Any()).Return(test.EksdRelease(), nil)
-	tt.mocks.provider.EXPECT().GenerateCAPISpecForUpgrade(tt.ctx, mCluster, wCluster, tt.clusterSpec, tt.clusterSpec.DeepCopy())
+	tt.mocks.provider.EXPECT().GenerateCAPISpecForUpgrade(tt.ctx, mCluster, wCluster, gomock.Any(), tt.clusterSpec.DeepCopy())
 	tt.mocks.client.EXPECT().ApplyKubeSpecFromBytesWithNamespace(tt.ctx, mCluster, test.OfType("[]uint8"), constants.EksaSystemNamespace).Times(2)
-	tt.mocks.provider.EXPECT().RunPostControlPlaneUpgrade(tt.ctx, tt.clusterSpec, tt.clusterSpec, wCluster, mCluster)
+	tt.mocks.provider.EXPECT().RunPostControlPlaneUpgrade(tt.ctx, gomock.Any(), tt.clusterSpec, wCluster, mCluster)
```

Comment:
curious why `Any`?

Reply:
For some reason, it believes the expected argument in index 1 is different from the actual value even if they appear to be the exact same.

Reply:
It thought the actual value was different from the expected value. I fixed it so that we can use tt.clusterSpec again.